components/script/dom/document.rs

```diff
@@ -2951,6 +2966,28 @@ impl DocumentMethods for Document {
 
     // https://html.spec.whatwg.org/multipage/#documentandelementeventhandlers
     document_and_element_event_handlers!();
+
+    event_handler!(fullscreenerror, GetOnfullscreenerror, SetOnfullscreenerror);
+
+    event_handler!(fullscreenchange, GetOnfullscreenchange, SetOnfullscreenchange);
+
+    fn FullscreenEnabled(&self) -> bool {
+        self.fullscreen_element.get().is_some()
+    }
+
+    fn Fullscreen(&self) -> bool {
+        self.FullscreenEnabled()
+    }
+
+    fn GetFullscreenElement(&self) -> Option<Root<Element>> {
+        self.fullscreen_element.get()
+    }
+
+    #[allow(unrooted_must_root)]
+    fn ExitFullscreen(&self) -> Rc<Promise> {
+        self.set_fullscreen_element(None);
+        Promise::new(self.global().r())
```

Comment:
Same promise comment. `set_fullscreen_element` should probably return the `Promise` value so we don't duplicate the code.